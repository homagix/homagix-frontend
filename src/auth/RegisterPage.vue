<script setup lang="ts">
import { ActionType, useStore } from "@/store"
import { ref } from "vue"
import { useRouter } from "vue-router"
import AppButton from "../components/AppButton.vue"

const router = useRouter()
const store = useStore()

let name = ref("")

async function register() {
  await store.dispatch(ActionType.REGISTER_USER, name.value)
  router.push("/my")
}

function cancel() {
  router.back()
}
</script>

<template>
  <h2 class="title is-4">Neu registrieren</h2>
  <p>
    Hier kannst du deinem Zugang einen Namen (z.B. deinen Vornamen oder auch einen Phantasienamen) geben. Wir sprechen
    dich dann k√ºnftig damit an.
  </p>
  <label>
    Name
    <input type="text" v-model="name" />
  </label>

  <div class="button-list">
    <AppButton id="register-button" @click="register">Registrieren</AppButton>
    <AppButton id="cancel-button" @click="cancel">Abbrechen</AppButton>
  </div>
</template>

<style lang="scss">
label {
  display: block;
}

.button-list {
  margin-top: 1rem;
}
</style>
