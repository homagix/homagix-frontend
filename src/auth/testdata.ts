import { vi } from "vitest"
import { createStore } from "vuex"
import { mount } from "@vue/test-utils"
import { createRouterMock } from "../mocks/router"
import type { User } from "."

export const token =
  "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjMiLCJmaXJzdE5hbWUiOiJhYmMiLCJpYXQiOjE2NDY5Nzg0NTUsImV4cCI6MTY0NzA2NDg1NX0.kClNCHxRnm_rvNpkdwdHgDsVebJfOzZ66Z_5epxkKNU"

export const user: User = { id: "123", name: "abc", accessCode: "456" }

export function mountComponent<C>(component: C) {
  const router = createRouterMock(component, ["/my", "/register"])
  const registerUserAction = vi.fn()
  const store = createStore({
    state() {
      return {}
    },
    actions: {
      REGISTER_USER: registerUserAction,
    },
  })
  const wrapper = mount(component, { global: { plugins: [store, router] } })
  return { wrapper, registerUserAction, router }
}
