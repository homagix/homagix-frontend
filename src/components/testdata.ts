import { Dish, Ingredient } from "@/types"

export const avocado = { id: "42", name: "Avocado", unit: "Stk" } as Ingredient
export const tomato = { id: "43", name: "Tomate", unit: "Stk" } as Ingredient

export const guacamole: Dish = {
  id: "5",
  name: "Guacamole",
  items: [
    { id: avocado.id, amount: 2 },
    { id: tomato.id, amount: 2 },
  ],
  alwaysOnList: false,
  images: ["guacamole.jpg"],
  isFavorite: false,
  isEditable: true,
}

export const tomatoSoup: Dish = {
  id: "7",
  name: "Tomatensuppe",
  items: [{ id: tomato.id, amount: 4 }],
  alwaysOnList: false,
  images: ["tomatoSoup.jpg"],
  isFavorite: false,
  isEditable: false,
}
