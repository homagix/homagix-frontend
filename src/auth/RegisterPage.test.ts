import { describe, expect, it, vi } from "vitest"
import RegisterPage from "@/auth/RegisterPage.vue"
import { mountComponent } from "./testdata"

describe("RegisterPage", () => {
  it("should render", () => {
    const { wrapper } = mountComponent(RegisterPage)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should return to the previous page if 'cancel' button is clicked", () => {
    const { wrapper, router } = mountComponent(RegisterPage)
    router.back = vi.fn()
    wrapper.find("#cancel-button").trigger("click")
    expect(router.back).toHaveBeenCalled()
  })

  it("should dispatch REGISTER_USER action if button is clicked", () => {
    const { wrapper, actionMocks } = mountComponent(RegisterPage)
    wrapper.find("input").setValue("luigi")
    wrapper.find("#register-button").trigger("click")
    expect(actionMocks.REGISTER_USER).toHaveBeenCalledOnce()
    expect(actionMocks.REGISTER_USER).toHaveBeenCalledWith(expect.any(Object), "luigi")
  })

  it("should got to /my page after registration", async () => {
    const { wrapper, router } = mountComponent(RegisterPage)
    router.push = vi.fn()
    await wrapper.find("#register-button").trigger("click")
    expect(router.push).toHaveBeenCalledWith("/my")
  })
})
