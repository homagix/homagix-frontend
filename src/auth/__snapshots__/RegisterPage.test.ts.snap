// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RegisterPage > should render 1`] = `
"<h2 class=\\"title is-4\\">Neu registrieren</h2>
<p> Hier kannst du deinem Zugang einen Namen (z.B. deinen Vornamen oder auch einen Phantasienamen) geben. Wir sprechen dich dann kÃ¼nftig damit an. </p>
<label> Name <input type=\\"text\\"></label>
<div class=\\"button-list\\">
  <app-button-stub id=\\"register-button\\"></app-button-stub>
  <app-button-stub id=\\"cancel-button\\"></app-button-stub>
</div>"
`;
